src-plugins/manualRegistration/manualRegistrationToolBox.cpp

```diff
@@ -332,45 +319,66 @@ void manualRegistrationToolBox::synchroniseMovingFuseView()
-    d->process->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
-    d->process->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
-    d->process->setParameter(d->transformType->itemData(d->transformType->currentIndex()).toInt());
-    d->process->update(itkProcessRegistration::FLOAT);
+            d->process = dynamic_cast<medAbstractRegistrationProcess*> (dtkAbstractProcessFactory::instance()->create("manualRegistration"));
 
-    medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            if(toolbox) // toolbox empty in Pipelines and not Registration workspace
+            {
+                toolbox->setProcess(d->process);
+            }
 
-    if(toolbox) // toolbox empty in Pipelines and not Registration workspace
-    {
-        toolbox->setProcess(d->process);
+            dtkSmartPointer<manualRegistration> process_Registration = dynamic_cast<manualRegistration* >(d->process.data());
+            process_Registration->SetFixedLandmarks(d->controller->getPoints_Fixed());
+            process_Registration->SetMovingLandmarks(d->controller->getPoints_Moving());
+            process_Registration->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
+            process_Registration->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
+            process_Registration->setParameter(transformChosen);
+
+            medRunnableProcess *runProcess = new medRunnableProcess;
+            runProcess->setProcess (d->process);
+            connect (runProcess, SIGNAL (success(QObject*)), this, SLOT(retrieveProcessOutputAndUpdateViews()));
+            this->addConnectionsAndStartJob(runProcess);
+        }
     }
+}
 
+void manualRegistrationToolBox::retrieveProcessOutputAndUpdateViews()
+{
     medAbstractData * newOutput = d->process->output();
 
     if(newOutput && newOutput->data())
     {
         d->output = newOutput;
 
-        medUtilities::setDerivedMetaData(d->output, (qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0)), "registered");
+        // Pipelines
+        if (d->bottomContainer && d->bottomContainer->view())
+        {
+            qobject_cast<medAbstractImageView*>(d->bottomContainer->view())->removeLayer(1);
+            qobject_cast<medAbstractImageView*>(d->bottomContainer->view())->addLayer(d->output);
+        }
 
-        qobject_cast<medAbstractImageView*>(d->bottomContainer->view())->removeLayer(1);
-        qobject_cast<medAbstractImageView*>(d->bottomContainer->view())->addLayer(d->output);
         synchroniseMovingFuseView();
 
-        setDisableSaveButtons(false);
+        d->process.releasePointer();
```

Comment:
Use ```d->process = 0``` to decrement the reference count. The ```releasePointer``` function releases the pointer without decreasing the reference count.